iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -578,9 +578,7 @@ private async Task HandleIncomingMessagesAsync()
             // We are intentionally not awaiting _deviceMessageReceivedListener callback.
             // This is a user-supplied callback that isn't required to be awaited by us. We can simply invoke it and continue.
             _ = _deviceMessageReceivedListener?.Invoke(message);
-            // Messages with QoS = 1 need to be Acknowledged otherwise it results in mismatched Ack to IoT Hub
-            // causing next message being replayed and all subsequent messages being queued.
-            await CompleteIncomingMessageAsync(message).ConfigureAwait(false);
```

Comment:
The comment states that this completion is required for QoS 1 messages which would otherwise cause messages to keep being replayed (if, for instance, the user doesn't explicitly mark the message as complete). Is the issue stemming from the fact that both the SDK and user are trying to complete the same message?

Reply:
Also, doesn't the "complete" API take in the message/lock token as the input? Does the internal queue synchronization not take into account the passed in lock token?

Reply:
Yes, I added this change a few months ago to fix M2M message failure after direct method. So, this change was specifically for HandleIncomingMethodPostAsync() and HandleIncomingTwinPatchAsync()

Reply:
So, my question is - is the previous comment added here invalid? Is the QoS 1 ack required only for methods and twin?

Reply:
Yes, the previous comment here is incorrect. The SDK trying to complete QoS 1 message internally is not needed.

Reply:
Alright. I'd recommend making a note to add the above module client scenarios so that we can have the tests validate the behavior for us.